torch/_utils.py

```diff
@@ -577,14 +577,20 @@ def _get_devices_properties(device_ids):
     return [_get_device_attr(lambda m: m.get_device_properties(i)) for i in device_ids]
 
 
+def is_xpu_available() -> bool:
+    return torch._C._is_xpu_available()
+
+
 def get_current_device_index() -> int:
-    r"""Checks if there are CUDA devices available and
-    returns the device index of the current default CUDA device.
-    Returns -1 in case there are no CUDA devices available.
+    r"""Checks if there are GPU devices available and
+    returns the device index of the current default GPU device.
+    Returns -1 in case there are no GPU devices available.
     Arguments: ``None``
     """
     if torch.cuda.device_count() > 0:
         return torch.cuda.current_device()
+    if is_xpu_available() and torch.xpu.device_count() > 0:  # type: ignore[attr-defined]
```

Comment:
Hi @bdhirsh, I would like to supplement XPU backend here. But hasattar(torch, "xpu") is not supported in JIT. So I have to register an API that can tell me whether torch_xpu_extension has been imported.
We would like to use XPUHooks to implement this. And another reason for XPUHooks is adding DLPack support in XPU backend. You can refer to (#82867)[https://github.com/pytorch/pytorch/pull/82867] for detail.

Reply:
Hey @guangyey, thanks for the context.

In the spirit of adding fewer backend-specific python API's - can change it to be generic, and take in a backend? Something like:

```
def is_backend_available("xpu")

// You can parse the dispatch key inside of here into `DispatchKey::XPU`, and forward it to your XPU hooks
torch._C._is_backend_available("xpu")
```